updatecli/updatecli.d/update-versions-config-yaml/update-versions-config.yaml

```diff
@@ -0,0 +1,102 @@
+name: '{{ .k8s_k3s.manifest }}'
+pipelineid: '{{ .k8s_k3s.pipelineid }}'
+
+sources:
+  rancher-go-mod-k8s:
+    name: 'Get K8s version in go.mod'
+    kind: golang/gomod
+    scmid: 'default'
+    spec:
+      module: 'k8s.io/kubernetes'
+    transformers:
+      - find: 'v\d+\.\d+'
+
+  k8s-release:
+    name: 'Get latest upstream K8s version for {{ source "rancher-go-mod-k8s" }} patch version'
+    kind: githubrelease
+    spec:
+      owner: '{{ .repos.k8s.owner }}'
+      repository: '{{ .repos.k8s.repo }}'
+      key: tagname
+      typefilter:
+        release: true
+      username: '{{ .scm.username }}'
+      token: '{{ requiredEnv .scm.token }}'
+      versionfilter:
+        kind: semver
+        pattern: '~{{ source "rancher-go-mod-k8s" }}'
```

Comment:
Does it guarantee to return the latest patch version?

Reply:
Yes, the `~` will only filter for the patch versions and by default updatecli will pick the latest patch version available (not in terms of release date, but in terms of version number, which is what we want here).